arch_ppc32.py

```diff
@@ -0,0 +1,43 @@
+from core import ADDR, REG
+from archutils import *
+
+
+BITNESS = 32
+ENDIANNESS = "big"
+
+ALL_REGS = {REG("r0"), REG("sp"), REG("rtoc")} | reg_range("r", 3, 31)
+
+
+def call_params(addr):
+    return reg_range("r", 3, 10)
+
+
+def param_filter(regs):
+    # Assume for now that all parameters will be passed in registers.
+    # This must be good enough to cover the most usage cases.
+    # The following cases will break it:
+    # - functions accepting more then 8 params
+    # - functions accepting floating-point params
+    # - variadic functions
+    return reg_continuous_subrange(regs, reg_range("r", 3, 10))
+
+
+def call_ret(addr):
+    return {REG("r3")}
```

Comment:
Btw, how longlong's and double's are returned?

Reply:
According to Apple's specs, any value larger than 4 bytes must be returned via memory. For that purpose, the left-most parameter (in other words, R3) must point to a memory location where the return value will be stored to.

The same rule is valid for more than 8 params and variadic functions. Params[8:] as well as variadic args must be passed on the stack.

Float & doubles will be returned in dedicated floating-point registers. I assume that floating-point isn't currently supported so I omitted them for the moment being.

Reply:
I see, sounds good.

> I assume that floating-point isn't currently supported

Yes, there's no any special support it.